packages/ember-glimmer/lib/utils/references.js

```diff
@@ -110,7 +110,7 @@ export class PropertyReference extends CachedReference { // jshint ignore:line
       _parentObjectTag.update(tagFor(parentValue));
     }
 
-    if (parentValue && typeof parentValue === 'object') {
+    if (typeof parentValue === 'object' && parentValue) {
```

Comment:
rather then coercion we can be explicit, the only case we care about here is `parentValue !== null`